packages/cli/src/controller/generate-controller.ts

```diff
@@ -221,65 +260,121 @@ function filterExistingFragments<T extends Fragment | ConstructorFragment>(
   return cleanFragments;
 }
 
-export function filterExistingMethods(
+export type ManifestExtractor<T> = (
+  dataSources: T,
+  casedInputAddress: string | undefined
+) => {
+  existingEvents: string[];
+  existingFunctions: string[];
+};
+
+export function filterExistingMethods<T>(
   eventFragments: Record<string, EventFragment>,
   functionFragments: Record<string, FunctionFragment>,
-  dataSources: EthereumDs[],
-  address: string | undefined
+  dataSources: T,
+  address: string | undefined,
+  extractor: ManifestExtractor<T>
 ): [Record<string, EventFragment>, Record<string, FunctionFragment>] {
+  const {existingEvents, existingFunctions} = extractor(dataSources, address && address.toLowerCase());
+
+  return [
+    filterExistingFragments<EventFragment>(eventFragments, existingEvents),
+    filterExistingFragments<FunctionFragment>(functionFragments, existingFunctions),
+  ];
+}
+
+export const yamlExtractor: ManifestExtractor<EthereumDs[]> = (dataSources, casedInputAddress) => {
   const existingEvents: string[] = [];
   const existingFunctions: string[] = [];
 
-  const casedInputAddress = address && address.toLowerCase();
-
   dataSources
-    .filter((d) => {
+    .filter((d: EthereumDs) => {
       if (casedInputAddress && d.options.address) {
-        return casedInputAddress.toLowerCase() === d.options.address.toLowerCase();
+        return casedInputAddress === d.options.address.toLowerCase();
       }
       return casedInputAddress === d.options.address || (!casedInputAddress && !d.options.address);
```

Comment:
This doesn't make much sense. The `casedInputAddress === d.options.address` outside of the if will always be false. 

Also why is `.toLowerCase()` removed here?